### PR TITLE
Radiation plugin: Fix checkpointing

### LINKED ISSUE #4397: radiation plugin does not produce openPMD output anymore
The current `dev` version of PIConGPU does not produce openPMD radiation output anymore. 
Most likely this is caused by #4387. A simulation that used the 2022-11-22 `dev` branched produced the openPMD files.

### PR DESCRIPTION
**Pull request by franzpoeschel**
Follow-up to https://github.com/ComputationalRadiationPhysics/picongpu/pull/4387
Fix #4397

### CODE REVIEW

**Comment by PrometheusPi** [comment]
There seems to have been a crash in the CI with some clang compiler - I restarted it.

**Comment by psychocoderHPC** [comment]
@franzpoeschel Are checkpoints with your fix always written as one file per checkpoint? This is a strong requirement for picongpu. Standard IO can be optionally in one file with a time series.

**Comment by franzpoeschel** [comment]
@psychocoderHPC yes, they are. With this fix, I don't even keep the Series open for checkpointing.
EDIT: Wait before merging, I'll actually have to check if I kept the filename the same for checkpoints

File: include/picongpu/plugins/radiation/Radiation.hpp [hunk 1]
```diff
@@ -282,9 +282,12 @@ namespace picongpu
...
                     if(isMaster)
                     {
-                        writeOpenPMDfile(
-                            tmp_result,
-                            restartDirectory + "/" + speciesName + std::string("_radRestart"));
+                        /*
+                         * No need to keep the Series open for checkpointing, so
+                         * just quickly open it here.
+                         */
+                        std::optional<::openPMD::Series> tmp;
```
Comment (franzpoeschel) [inline include/picongpu/plugins/radiation/Radiation.hpp:302]: See here @psychocoderHPC 
This is the Series for checkpointing, just quickly opened here and then closed again

**Comment by franzpoeschel** [comment]
It's good that you asked René. The parameter `--radiation.openPMDSuffix` is read by both the checkpointing and regular output routines, while it should only be read by regular output. Will have to fix.

**Comment by franzpoeschel** [comment]
Alright, these are now the new parameters:
```
  --e_radiation.openPMDSuffix arg (=_%T_0_0_0.h5)
                                        Suffix for openPMD filename extension 
                                        and iteration expansion pattern.
  --e_radiation.openPMDCheckpointExtension arg (=h5)
                                        Filename extension for openPMD 
                                        checkpoints.
  --e_radiation.openPMDConfig arg (={}) JSON/TOML configuration for 
                                        initializing openPMD.
  --e_radiation.openPMDCheckpointConfig arg (={})
                                        JSON/TOML configuration for 
                                        initializing openPMD checkpointing.
```

No --restartConfig parameter for now, since noone will use it anyway.

**Review by psychocoderHPC — APPROVED** [review]

**Review by PrometheusPi — APPROVED** [review]
Tested in - all files are created

**Pull request merged**